core_plugins/reporting/src/index.ts

```diff
@@ -1,29 +1,29 @@
-import { KibanaFunctionalPlugin } from 'kbn-types';
+import { KibanaPluginConfig } from 'kbn-types';
 import { XPackPluginType } from 'kibana-plugin-xpack';
 import { ReportingConfig } from './ReportingConfig';
 
-export const configPath = ['xpack', 'reporting'];
-
-export const dependencies = ['xpack'];
-
-export const plugin: KibanaFunctionalPlugin<XPackPluginType> = (
-  kibana,
-  dependencies
-) => {
-  const { xpack } = dependencies;
-
-  const log = kibana.logger.get();
-
-  const config$ = kibana.config.createIfExists(ReportingConfig);
-
-  // just an example
-  const isEnabled$ = config$.map(config => config && config.enabled);
-
-  isEnabled$.subscribe(isEnabled => {
-    log.info(`reporting enabled? [${isEnabled}]`);
-  });
-
-  xpack.config$.subscribe(xpackConfig => {
-    log.info(`xpack polling frequency: [${xpackConfig.pollingFrequency}]`);
-  });
-};
+export const plugin: KibanaPluginConfig<XPackPluginType> = {
+  configPath: ['xpack', 'reporting'],
+  dependencies: ['xpack'],
+  plugin: (
```

Comment:
__optional nit:__ can be just `plugin(kibana, dependencies) {` I think, but we can leave it to the PR where `prettier` is configured for the core plugins folder.

Reply:
Yep, will fix in the prettier PR